backend/remote-state/s3/backend_test.go

```diff
@@ -249,6 +249,63 @@ func TestBackendExtraPaths(t *testing.T) {
 	}
 }
 
+func TestKeyEnv(t *testing.T) {
+	testACC(t)
+	bucketName := fmt.Sprintf("terraform-remote-s3-test-%x", time.Now().Unix())
+	keyName := "tfstate"
```

Comment:
These test cases are good, but let's go back to having a keyName with a `/` to make sure it's still supported after we introduce them into the prefix.